### ISSUE TITLE
When I press tab to complete code or get any hint, the ipython have nothing interaction

### ISSUE THREAD

**Issue by jkfx** [ipython/ipython#12776]
When I press Tab to complete code or get any hint, the ipython have nothing interaction, however, I continue inputing my code or I press directly Enter, then the iPython fall crash and output:

```cmd
Python 3.7.9 (default, Aug 31 2020, 17:10:11) [MSC v.1916 64 bit (AMD64)]
Type 'copyright', 'credits' or 'license' for more information
IPython 7.19.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import matplot
Traceback (most recent call last):
  File "d:\programming\miniconda3\envs\jkfx\lib\site-packages\IPython\terminal\ptutils.py", line 113, in get_completions
    yield from self._get_completions(body, offset, cursor_position, self.ipy_completer)
  File "d:\programming\miniconda3\envs\jkfx\lib\site-packages\IPython\terminal\ptutils.py", line 129, in _get_completions
    for c in completions:
  File "d:\programming\miniconda3\envs\jkfx\lib\site-packages\IPython\core\completer.py", line 438, in _deduplicate_completions
    completions = list(completions)
  File "d:\programming\miniconda3\envs\jkfx\lib\site-packages\IPython\core\completer.py", line 1818, in completions
    for c in self._completions(text, offset, _timeout=self.jedi_compute_type_timeout/1000):
  File "d:\programming\miniconda3\envs\jkfx\lib\site-packages\IPython\core\completer.py", line 1862, in _completions
    full_text=full_text, cursor_line=cursor_line, cursor_pos=cursor_column)
  File "d:\programming\miniconda3\envs\jkfx\lib\site-packages\IPython\core\completer.py", line 2030, in _complete
    cursor_pos, cursor_line, full_text)
  File "d:\programming\miniconda3\envs\jkfx\lib\site-packages\IPython\core\completer.py", line 1374, in _jedi_matches
    text[:offset], namespaces, column=cursor_column, line=cursor_line + 1)
  File "d:\programming\miniconda3\envs\jkfx\lib\site-packages\jedi\api\__init__.py", line 726, in __init__
    project=Project(Path.cwd()), **kwds)
TypeError: __init__() got an unexpected keyword argument 'column'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "d:\programming\miniconda3\envs\jkfx\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "d:\programming\miniconda3\envs\jkfx\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "D:\Programming\Miniconda3\envs\jkfx\Scripts\ipython.exe\__main__.py", line 7, in <module>
    sys.exit(start_ipython())
  File "d:\programming\miniconda3\envs\jkfx\lib\site-packages\IPython\__init__.py", line 126, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "d:\programming\miniconda3\envs\jkfx\lib\site-packages\traitlets\config\application.py", line 845, in launch_instance
    app.start()
  File "d:\programming\miniconda3\envs\jkfx\lib\site-packages\IPython\terminal\ipapp.py", line 356, in start
    self.shell.mainloop()
  File "d:\programming\miniconda3\envs\jkfx\lib\site-packages\IPython\terminal\interactiveshell.py", line 564, in mainloop
    self.interact()
  File "d:\programming\miniconda3\envs\jkfx\lib\site-packages\IPython\terminal\interactiveshell.py", line 547, in interact
    code = self.prompt_for_code()
  File "d:\programming\miniconda3\envs\jkfx\lib\site-packages\IPython\terminal\interactiveshell.py", line 475, in prompt_for_code
    **self._extra_prompt_options())
  File "d:\programming\miniconda3\envs\jkfx\lib\site-packages\prompt_toolkit\shortcuts\prompt.py", line 1013, in prompt
    return self.app.run(set_exception_handler=set_exception_handler)
  File "d:\programming\miniconda3\envs\jkfx\lib\site-packages\prompt_toolkit\application\application.py", line 817, in run
    self.run_async(pre_run=pre_run, set_exception_handler=set_exception_handler)
  File "d:\programming\miniconda3\envs\jkfx\lib\asyncio\base_events.py", line 587, in run_until_complete
    return future.result()
  File "d:\programming\miniconda3\envs\jkfx\lib\site-packages\prompt_toolkit\application\application.py", line 783, in run_async
    return await _run_async2()
  File "d:\programming\miniconda3\envs\jkfx\lib\site-packages\prompt_toolkit\application\application.py", line 771, in _run_async2
    await self.cancel_and_wait_for_background_tasks()
  File "d:\programming\miniconda3\envs\jkfx\lib\site-packages\prompt_toolkit\application\application.py", line 872, in cancel_and_wait_for_background_tasks
    await task
  File "d:\programming\miniconda3\envs\jkfx\lib\site-packages\prompt_toolkit\buffer.py", line 1854, in new_coroutine
    await coroutine(*a, **kw)
  File "d:\programming\miniconda3\envs\jkfx\lib\site-packages\prompt_toolkit\buffer.py", line 1684, in async_completer
    document, complete_event
  File "d:\programming\miniconda3\envs\jkfx\lib\site-packages\prompt_toolkit\completion\base.py", line 270, in get_completions_async
    document, complete_event
  File "d:\programming\miniconda3\envs\jkfx\lib\site-packages\prompt_toolkit\completion\base.py", line 196, in get_completions_async
    for item in self.get_completions(document, complete_event):
  File "d:\programming\miniconda3\envs\jkfx\lib\site-packages\IPython\terminal\ptutils.py", line 116, in get_completions
    exc_type, exc_value, exc_tb = sys.exc_info()
NameError: name 'sys' is not defined

If you suspect this is an IPython 7.19.0 bug, please report it at:
    https://github.com/ipython/ipython/issues
or send an email to the mailing list at ipython-dev@python.org

You can print a more detailed traceback right now with "%tb", or use "%debug"
to interactively debug it.

Extra-detailed tracebacks for bug-reporting purposes can be enabled via:
    %config Application.verbose_crash=True
```

I don't know the causion of problem whether CPU is AMD(Ryzen 4800U).

**Comment by nneibaue** [comment]
I am getting the exact same behavior. Running on MacOSX Big Sur 11.1. Tried at system level and inside virtual environment

**Comment by bwh1te** [comment]
Same on MacOS Mojave 10.14.6, iterm2, python 3.9.0 (through the pyenv based virtualenv)

**Comment by cesaralba** [comment]
Same here, Fedora 33, python 3.9.1 using virtual environment with pip. I sent a crash report.

**Comment by bwh1te** [comment]
**TL;DR:** `pip install jedi==0.17.2` did the trick for me.

In fact, wrong signature and of course non-pinning the dependencies...

`IPython/core/completer.py`:
```
class IPCompleter(Completer):

    def _jedi_matches(self, cursor_column:int, cursor_line:int, text:str):
        interpreter = jedi.Interpreter(
            text[:offset], namespaces, column=cursor_column, line=cursor_line + 1)
```

`jedi/api/__init__.py`:
```
class Interpreter(Script):

    def __init__(self, code, namespaces, **kwds):
        ...
        super().__init__(code, environment=environment,
                         project=Project(Path.cwd()), **kwds)

class Script:

    def __init__(self, code=None, *, path=None, environment=None, project=None):
        ... 
```

**Comment by bwh1te** [comment]
Oh, looks like it's already in master https://github.com/ipython/ipython/commit/90fb12b53e52314979167098de1c47f75f8fb14e#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7
So I hope it will be released soon.
